docs/reference/language/context.md

```diff
@@ -25,38 +25,180 @@ in some places that are also aware of their location in the document:
-function.
+## Behavior of the context keyword
+Style properties frequently change within a document, for example by applying set 
+rules. To retrieve such poperties in a consistent way, one must first specify 
+the context where the query is to be executed. This is the purpose of the 
+`context` keyword. Once the context has been fixed, the property information 
+is available through a simple field access syntax. For example, `text.lang` 
+asks for the current language setting. In its simplest form, the `context` 
+keyword refers to "right here":
 
 ```example
 #set text(lang: "de")
+// query the language setting "here"
 #context text.lang
 ```
 
-As explained above, a context expression is reactive to the different
-environments it is placed into. In the example below, we create a single context
-expression, store it in the `value` variable and use it multiple times. Each use
-properly reacts to the current surroundings.
+Note that calling `#text.lang` directly would be an error, because the request
+cannot be answered without knowledge of the context. The field names supported 
+by a given element function always correspond to the named parameters documented 
+on each element's page.
+
+Moreover, some functions, such as [`to-absolute`]($length.to-absolute) 
+and [`counter.display`]($counter.display) are only applicable in a context, 
+because their results depend on the current settings of style properties. 
+When another function `foo()` calls a context-dependent function, it becomes 
+itself context-dependent:
+
+```example
+#let foo() = 1em.to-absolute()
+#context {
+  // foo() cannot be called outside of a context
+  foo() == text.size
+}
+```
+
+When a property is changed, the response to the query changes accordingly: 
 
 ```example
-#let value = context text.lang
-#value
+#set text(lang: "en")
+#context text.lang
 
 #set text(lang: "de")
-#value
+#context text.lang
+```
+
+The output of a `#context ...` call is static in the form of opaque 
+[`content`]. Write access to context output is prohibited, as it would 
+often result in invalid code: If the context changes between read and 
+write, overwriting a property would cause an inconsistent system state. 
+In fact, context-dependent property fields are immutable constants even 
+within the context itself:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #text.lang
+]
+```
+
+Both calls have the same output 'en', because `text.lang` is assigned
+upon entry in the context and remains constant until the end of its scope 
+(the closing `]`). It does not "see" the `#set text(lang: "fr")` before 
+call 2. Compare this to the previous example: there we got two different 
+results because we created two different contexts.
+
+However, immutability only applies to the property fields themselves. 
+Content creation instructions within a context _do_ see the effect of 
```

Comment:
"Content creation instructions" is a bit of a mouthful, maybe just "content expressions"

Reply:
I feel that "Content creation instructions" is a less technical term, so maybe better for newcomers.

Reply:
"Content expressions" is indeed bad. I think my mind was thinking about "context expressions" when I wrote this. But the new version in the latest commit reads well.